tests/test-sources/plugins/utils/spectre.nix

```
@@ -0,0 +1,272 @@
+{pkgs, ...}: let
+  # Fails on darwin with: `module 'plenary.job' not found`
+  enable = !pkgs.stdenv.isDarwin;
+in {
+  empty = {
+    plugins.spectre.enable = enable;
+  };
+
+  example = {
+    plugins.spectre = {
+      inherit enable;
+
+      settings = {
+        live_update = true;
+        is_insert_mode = false;
+        find_engine = {
+          rg = {
+            cmd = "rg";
+            args = [
+              "--color=never"
+              "--no-heading"
+              "--with-filename"
+              "--line-number"
+              "--column"
+            ];
+            options = {
+              ignore-case = {
+                value = "--ignore-case";
+                icon = "[I]";
+                desc = "ignore case";
+              };
+              hidden = {
+                value = "--hidden";
+                desc = "hidden file";
+                icon = "[H]";
+              };
+              line = {
+                value = "-x";
+                icon = "[L]";
+                desc = "match in line";
+              };
+              word = {
+                value = "-w";
+                icon = "[W]";
+                desc = "match in word";
+              };
+            };
+          };
+        };
+        default = {
+          find = {
+            cmd = "rg";
+            options = ["word" "hidden"];
+          };
+          replace = {
+            cmd = "sed";
+          };
+        };
+      };
+    };
+  };
+
```

```suggestion

  no-packages = {
    plugins.spectre = {
      inherit enable;
      
      findPackage = null;
      replacePackage = null;

      settings = {
        default = {
          find.cmd = "rg";
          replace.cmd = "sed";
        };
      };
    };
  };
```

Does it make sense to have a test with non-default values for `{find,replace}Package`?

---

In this case, yes why not ! done